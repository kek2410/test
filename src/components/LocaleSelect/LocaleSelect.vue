<template>
  <v-col class="d-flex" cols="12" sm="2">
    <v-select
      v-model="selectValue"
      :items="items"
      item-text="text"
      item-value="value"
      label="Select"
      outlined
      dense
      id="testtest"
      name="testtest"
      @change="changeSelect"
    ></v-select>
    <v-btn @click="changeSelect">
      Action
    </v-btn>
    <button class="test111" @click="changeSelect">aa</button>
  </v-col>
</template>

<script lang="ts">
import { Vue, Component } from "vue-property-decorator";
import { State, Action } from "vuex-class";
// import { mutations } from "../../store/modules/Login/mutations";

const namespace = "login";

@Component({})
export default class LocaleSelect extends Vue {
  items = [
    { text: "대한민국", value: "ko" },
    { text: "English", value: "en" },
    { text: "日本語", value: "ja" }
  ];

  selectValue = "";

  @State(namespace) login!: any;
  @Action("setLocale", { namespace }) setLocale: any;

  changeSelect(val: any): void {
    this.setLocale(val);
    // this.$store.commit("login/setLocale", val);
  }

  created() {}

  mounted() {
    this.selectValue = this.login.locale;
  }
}
</script>
