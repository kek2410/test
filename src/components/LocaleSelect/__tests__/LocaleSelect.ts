import { mount, shallowMount, createLocalVue } from "@vue/test-utils";
// import { mount, createLocalVue } from "@vue/test-utils";
import LocaleSelect from "../LocaleSelect.vue";
import Vuex from "vuex";
import Vuetify from "vuetify";
// import store from "@/store";
import i18n from "@/locale";
import VueI18n from "vue-i18n";
import flushPromises from "flush-promises";

// import { VSelect } from "vuetify/lib";
// import { actions } from "@/store/modules/Login/actions";
// import { login } from "@/store/modules/Login";

const localVue = createLocalVue();
localVue.use(Vuex);
localVue.use(VueI18n);
localVue.use(Vuetify);

describe("LocaleSelect.vue!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!", (): void => {
  let mutations: any;
  let actions: any;
  let state: any;
  let store: any;
  let vuetify: any;

  beforeEach((): void => {
    vuetify = new Vuetify();

    state = {
      userInfo: {},
      locale: "ko"
    };

    mutations = {
      setLocale: jest.fn()
    };

    actions = {
      setLocale: jest.fn()
    };

    store = new Vuex.Store({
      modules: {
        login: {
          namespaced: true,
          state,
          mutations,
          actions
        }
      },
      strict: true
    });
  });

  it("LocaleSelect.vue", (): void => {
    const wrapper = mount(LocaleSelect, {
      localVue,
      store,
      vuetify,
      i18n
    });
    expect(wrapper.html()).toMatchSnapshot();

    const wr = shallowMount(LocaleSelect, {
      localVue,
      store,
      vuetify,
      i18n
    });
    expect(wr.html()).toMatchSnapshot();
  });

  it("selectMutation.vue", async (): Promise<any> => {
    // const wrapper = mount(LocaleSelect, { localVue, vuetify, store });
    const wr = mount(LocaleSelect, {
      localVue,
      store,
      vuetify,
      i18n
    });

    // console.log(wr.classes());
    const sb = wr.find(".v-select");

    sb.trigger("change", { itemvalue: "ja" });

    await flushPromises();

    expect(actions.setLocale).toHaveBeenCalled();

    // console.log(wr.find(".v-select").html());

    // expect(wr.classes()).toContain("div");

    // expect(wr.find(VSelect).exists()).toBe(true);
    // expect(wr.find({ name: "testtest" }).exists()).toBe(true);

    // console.log("!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!", sb);

    // const event = jest.fn();
    // const sb2 = wr.find(".v_btn");
    // wr.vm.$on("changeSelect", event);
    // expect(event).toHaveBeenCalledTimes(0);
    // sb2.trigger("click");
    // expect(event).toHaveBeenCalledTimes(1);

    // const sbs = sb.findAll("selection");
    // sb.trigger("change");
    // const event = jest.fn();
    // wr.vm.$on("changeSelect", event);

    // wr.find("#testtest").trigger("change");
    // await wr.vm.$nextTick();

    // console.log(wr.vm.selectValue);

    // expect(event).toHaveBeenCalledTimes(1);

    // expect(mutations.setLocale).toHaveBeenCalledWith({}, { value: "ko" });

    // console.log(store);

    // console.log(store.getters["login/getLocale"]);

    // const lc = store.login.state.locale ;

    // console.log(sb);
    // console.log(sbs.length);

    // expect(1).toBe(1);

    // sb.trigger("change", { value: "ja" });

    // sb.at(1).setSelected();
    // expect(state.locale).toBe("ja");

    // const change = jest.fn();
    // wr.v;
  });
});
