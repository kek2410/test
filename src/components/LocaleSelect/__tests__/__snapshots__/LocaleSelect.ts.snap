// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocaleSelect.vue!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! LocaleSelect.vue 1`] = `
<div class="d-flex col-sm-2 col-12">
  <div class="v-input v-input--dense theme--light v-text-field v-text-field--enclosed v-text-field--outlined v-select">
    <div class="v-input__control">
      <div role="button" aria-haspopup="listbox" aria-expanded="false" aria-owns="list-6" class="v-input__slot">
        <fieldset aria-hidden="true">
          <legend style="width: 0px;"><span>â€‹</span></legend>
        </fieldset>
        <div class="v-select__slot"><label for="testtest" class="v-label theme--light" style="left: 0px; position: absolute;">Select</label>
          <div class="v-select__selections"><input name="testtest" id="testtest" readonly="readonly" type="text" aria-readonly="true"></div>
          <div class="v-input__append-inner">
            <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon notranslate mdi mdi-menu-down theme--light"></i></div>
          </div>
        </div>
        <div class="v-menu">
          <div class="v-menu__content theme--light " style="max-height: 304px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"></div>
        </div>
      </div>
      <div class="v-text-field__details">
        <div class="v-messages theme--light"><span name="message-transition" tag="div" class="v-messages__wrapper"></span></div>
      </div>
    </div>
  </div> <button type="button" class="v-btn v-btn--contained theme--light v-size--default"><span class="v-btn__content">
    Action
  </span></button> <button class="test111">aa</button>
</div>
`;

exports[`LocaleSelect.vue!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! LocaleSelect.vue 2`] = `
<v-col-stub cols="12" sm="2" tag="div" class="d-flex">
  <v-select-stub errorcount="1" errormessages="" messages="" rules="" successmessages="" value="" appendicon="$dropdown" backgroundcolor="" dense="true" id="testtest" label="Select" loaderheight="2" clearicon="$clear" outlined="true" type="text" valuecomparator="function deepEqual(a, b) {
  if (a === b) return true;

  if (a instanceof Date &amp;&amp; b instanceof Date) {
    // If the values are Date, they were convert to timestamp with getTime and compare it
    if (a.getTime() !== b.getTime()) return false;
  }

  if (a !== Object(a) || b !== Object(b)) {
    // If the values aren't objects, they were already checked for equality
    return false;
  }

  var props = Object.keys(a);

  if (props.length !== Object.keys(b).length) {
    // Different number of props, don't bother to check
    return false;
  }

  return props.every(function (p) {
    return deepEqual(a[p], b[p]);
  });
}" nodatatext="$vuetify.noDataText" items="[object Object],[object Object],[object Object]" itemcolor="primary" itemdisabled="disabled" itemtext="text" itemvalue="value" menuprops="[object Object]" name="testtest"></v-select-stub>
  <v-btn-stub tag="button" activeclass="" type="button">
    Action
  </v-btn-stub> <button class="test111">aa</button>
</v-col-stub>
`;
